distributed/http/worker/prometheus/core.py

```diff
@@ -161,6 +146,128 @@ def collect(self):
+        """SpillBuffer-specific metrics.
+
+        Additionally, you can obtain derived metrics as follows:
+
+        cache hit ratios:
+          by keys  = spill_count_total.memory_read / (spill_count_total.memory_read + spill_count_total.disk_read)
+          by bytes = spill_bytes_total.memory_read / (spill_bytes_total.memory_read + spill_bytes_total.disk_read)
+
+        mean times per key:
+          pickle   = spill_time_total.pickle     / spill_count_total.disk_write
+          write    = spill_time_total.disk_write / spill_count_total.disk_write
+          unpickle = spill_time_total.unpickle   / spill_count_total.disk_read
+          read     = spill_time_total.disk_read  / spill_count_total.disk_read
+
+        mean bytes per key:
+          write    = spill_bytes_total.disk_write / spill_count_total.disk_write
+          read     = spill_bytes_total.disk_read  / spill_count_total.disk_read
+
+        mean bytes per second:
+          write    = spill_bytes_total.disk_write / spill_time_total.disk_write
+          read     = spill_bytes_total.disk_read  / spill_time_total.disk_read
+        """
+        try:
+            get_metrics = self.server.data.get_metrics  # type: ignore
+        except AttributeError:
+            return  # spilling is disabled
+        metrics = get_metrics()
+
+        total_bytes = CounterMetricFamily(
+            self.build_name("spill_bytes_total"),
+            "Total size of memory and disk accesses caused by managed data "
+            "since the latest worker restart",
+            labels=["event"],
+        )
+        # Note: memory_read is used to calculate cache hit ratios (see docstring)
+        for k in ("memory_read", "disk_read", "disk_write"):
+            total_bytes.add_metric([k], metrics[f"{k}_bytes_total"])
+
+        total_counts = CounterMetricFamily(
+            self.build_name("spill_count_total"),
+            "Total number of memory and disk accesses caused by managed data "
+            "since the latest worker restart",
+            labels=["event"],
+        )
+        # Note: memory_read is used to calculate cache hit ratios (see docstring)
+        for k in ("memory_read", "disk_read", "disk_write"):
+            total_counts.add_metric([k], metrics[f"{k}_count_total"])
+
+        total_times = CounterMetricFamily(
+            self.build_name("spill_time_total"),
+            "Total time spent spilling/unspilling since the latest worker restart",
+            labels=["event"],
+        )
+        for k in ("pickle", "disk_write", "disk_read", "unpickle"):
+            total_times.add_metric([k], metrics[f"{k}_time_total"])
+        yield total_times
+
+        max_times = GaugeMetricFamily(
+            self.build_name("spill_time_per_key_max"),
+            "Maximum time spent spilling/unspilling a single key "
+            "since the latest worker restart",
+            labels=["event"],
+        )
+        for k in ("pickle", "disk_write", "disk_read", "unpickle"):
```

Comment:
Notably the sum of these does *not* show the longest the event loop has been unresponsive due to spilling, because multiple spills and unspills can go in sequence without releasing the event loop:
- never released, when target threshold is exceeded (see `zict.Buffer.__setitem__` -> `zict.LRU.__setitem__`)
- never released, when retrieving the inputs for a task (see `Worker._prepare_args_for_execution`). You can read how long it took in total on `TaskState.startsstops`.
- released every 0.5s or the time it takes for a single key, whatever is slower, when the spill threshold is exceeded (see `WorkerMemoryManager._maybe_spill`)